invenio_stats/aggregations.py

```diff
@@ -292,20 +292,46 @@ def _split_date_range(self, lower_limit, upper_limit):
         res[dt_key] = upper_limit
         return res
 
+    def _get_stats_done(self, index_name, timestamp):
+        """Retrieve the statistics that have already been inserted."""
+        if not dsl.Index(index_name, using=self.client).exists():
+            return {}
+
+        my_query = (
+            dsl.Search(
+                using=self.client,
+                index=index_name,
+            )
+            .filter(
+                "term",
+                timestamp=timestamp,
+            )
+            .extra(size=self.max_bucket_size)
+            .source(includes=["count"])
+            .execute()
+        )
+        # Putting them into a hash for faster checking
+        already_done_hash = {}
+        for entry in my_query:
+            # Let's remove the date from the id
+            already_done_hash[entry.meta["id"][:-11]] = entry.get("count")
+
+        return already_done_hash
+
     def agg_iter(self, dt):
         """Aggregate and return dictionary to be indexed in the search engine."""
         rounded_dt = format_range_dt(dt, self.interval)
         self.agg_query = (
             dsl.Search(using=self.client, index=self.event_index).filter(
-                "range",
                 # Filter for the specific interval (hour, day, month)
-                timestamp={"gte": rounded_dt, "lte": rounded_dt},
+                "term",
+                timestamp=rounded_dt,
             )
             # we're only interested in the aggregated results but not the search hits,
             # so we tell the search to ignore them to save some bandwidth
             .extra(size=0)
         )
-
+        index_name = prefix_index("stats-{0}-{1}".format(self.event, rounded_dt[0:7]))
```

Comment:
minor: could you add a comment with an example index name to which this evaluates to? would be easier to read

Reply:
I've removed that line. Thanks!